muon/_prot/preproc.py

```diff
@@ -193,7 +194,9 @@ def dsb(
     return toreturn
 
 
-def clr(adata: AnnData, inplace: bool = True) -> Union[None, AnnData]:
+def clr(
+    adata: AnnData, inplace: bool = True, flavor: Optional[Literal["seurat"]] = None
+) -> Union[None, AnnData]:
```

Comment:
Please add documentation for the new parameter.

Reply:
Should the argument be called `flavor` or wouldn't be `axis` more suitable? `flavor="seurat"` is really only understandable to someone that knows Seurat (well), IMO.

Reply:
`flavor` is consistent with Scanpy, so I would be in favor of `flavor` (with a short explanation of what each flavor does). @gtca, what do you think?

Reply:
Yes, this is why I chose `flavor`. However, here, it is not a Seurat algorithm/approach to normalizing the data (as it is, for example, the case for dispersion [here](https://github.com/theislab/scanpy/blob/4dd8de9e355ce8d59009c15522670a3d0970462c/scanpy/preprocessing/_highly_variable_genes.py#L177). It's simply CLR applied along a different axis. Hence my change of heart.

Reply:
Indeed, `flavor` might be more consistent with `scanpy`. However it seems that the truly _Seurat_ flavor would be [slightly different](https://github.com/satijalab/seurat/blob/4e868fcde49dc0a3df47f94f5fb54a421bfdf7bc/R/preprocessing.R#L2390) from just another axis as it is now. So maybe we should keep `flavor='seurat'` but also change the behaviour accordingly? 
I'll also add a few comments on that below in the general discussion.

Another point is whether we should include a default flavor (rather than `None` as it is now)? See e.g. [`sc.pp.highly_variable_genes`](https://scanpy.readthedocs.io/en/stable/api/scanpy.pp.highly_variable_genes.html).

Reply:
In Seurat, there's a `margin` argument, and [by default it is 1 for CLR](https://github.com/satijalab/seurat/blob/4e868fcde49dc0a3df47f94f5fb54a421bfdf7bc/R/preprocessing.R#L2363) vs 2 for other normalisation methods. [In the latest vignettes](https://satijalab.org/seurat/articles/multimodal_vignette.html) CLR is used with `margin=2` so if we don't want do be opinionated here, we might need both arguments in the end. ..

Reply:
> Indeed, `flavor` might be more consistent with `scanpy`. However it seems that the truly _Seurat_ flavor would be [slightly different](https://github.com/satijalab/seurat/blob/4e868fcde49dc0a3df47f94f5fb54a421bfdf7bc/R/preprocessing.R#L2390) from just another axis as it is now. So maybe we should keep `flavor='seurat'` but also change the behaviour accordingly?

Sorry, can you ellaborate on the difference you see between this implementation and this Python implementation? The only thing I am aware of is the subsetting of `x`, i.e. `x[x > 0]`. TBH, I do not understand why it is needed, since all entries of `x` are be non-negative and zero entries do not cause problems since `log1p` is used. Or am I missung something?

> Another point is whether we should include a default flavor (rather than `None` as it is now)? See e.g. [`sc.pp.highly_variable_genes`](https://scanpy.readthedocs.io/en/stable/api/scanpy.pp.highly_variable_genes.html).

I set the default value to `None` to use the previous implementation as default. Not sure what a suitable and descriptive value would be. Would be an argument to use `axis`, IMO.

> In Seurat, there's a `margin` argument, and [by default it is 1 for CLR](https://github.com/satijalab/seurat/blob/4e868fcde49dc0a3df47f94f5fb54a421bfdf7bc/R/preprocessing.R#L2363) vs 2 for other normalisation methods. [In the latest vignettes](https://satijalab.org/seurat/articles/multimodal_vignette.html) CLR is used with `margin=2` so if we don't want do be opinionated here, we might need both arguments in the end...

So this would again be an argument for calling the argument `axis`, no?

Reply:
My understanding at the moment is that the `seurat` flavor is using `log1p` instead of `log`. I guess we can call your feature `axis`, and implement a second argument `flavor` for choosing the canonical or Seurat-flavored CLR. But you don't have to do that in this PR, I guess, just `axis` should be fine.